web/src/stream_list_sort.ts

```diff
@@ -71,22 +74,32 @@ export function has_recent_activity(sub) {
     return stream_topic_history.stream_has_topics(sub.stream_id) || sub.newly_subscribed;
 }
 
-export function has_recent_activity_but_muted(sub) {
+export function has_recent_activity_but_muted(sub: StreamSubscription): boolean {
     return has_recent_activity(sub) && sub.is_muted;
 }
 
-export function sort_groups(streams, search_term) {
-    const stream_id_to_name = (stream) => sub_store.get(stream).name;
+export function sort_groups(
+    stream_ids: number[],
+    search_term: string,
+): {
+    same_as_before: boolean;
+    pinned_streams: number[];
+    normal_streams: number[];
+    dormant_streams: number[];
+    muted_pinned_streams: number[];
+    muted_active_streams: number[];
+} {
+    const stream_id_to_name = (stream_id: number): string => sub_store.get(stream_id)!.name;
     // Use -, _ and / as word separators apart from the default space character
     const word_separator_regex = /[\s/_-]/;
-    streams = util.filter_by_word_prefix_match(
-        streams,
+    stream_ids = util.filter_by_word_prefix_match(
+        stream_ids,
```

Comment:
As a style strategy, it'd be a lot more readable to review this with a prep commit that renames the variable to `stream_id`  with a commit message explaining why that's correct, and then the main typescript conversion commit can just do that.

Reply:
Did that